### PR TITLE
Refactor the configuration into one package

### PR DESCRIPTION
**Pull request by oleorhagen**
See commit msg

### CODE REVIEW

**Comment by oleorhagen** [comment]
Sorry, did not get around it :)

File: conf/config.go [hunk 1]
```diff
@@ -127,6 +127,80 @@ type DualRootfsDeviceConfig struct {
...
+// MenderServer is a placeholder for a full server definition used when
+// multiple servers are given. The fields corresponds to the definitions
+// given in MenderConfig.
+type MenderServer struct {
+	ServerURL string
+	// TODO: Move all possible server specific configurations in
+	//       MenderConfig over to this struct. (e.g. TenantToken?)
+}
+
+type Config struct {
```
Comment (lluiscampos) [inline conf/config.go:197]: In this package now, `conf.Config` doesn't really describe what this struct holds. When it was `client.Config` made more sense.

If it is not too much work, can we rename to something explicit like `conf.ClientConfig` or `conf.HttpConfig`?

**Review by lluiscampos — APPROVED** [review]
I think we could rename the struct (see below), but being this soon legacy code I won't insist in beatifications. Looks good!

**Review by lluiscampos — APPROVED** [review]

File: conf/config.go (hunk 1 above)
Comment (oleorhagen) [reply conf/config.go]: Went for `HttpConfig`

**Pull request merged**